### PR TITLE
Add CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by dinatale2**
Provide contributing guidelines for new contributions
to mpiFileUtils.

Signed-off-by: Giuseppe Di Natale <dinatale2@llnl.gov>

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+* Please try to limit pull requests to a single commit which resolves
+one issue.
+* Make sure your commit messages must have a `Signed-off-by` line. See
+[Contributor's Declaration](#contributors-declaration) for more information.
+* For large pull requests consider structuring your changes as a stack of
+logically independent patches which build on each other.  This makes large
+changes easier to review and approve which speeds up the merging process.
+* Try to keep pull requests simple. Simple code with comments is much easier
+to review and approve.
+* Test cases should be provided when appropriate.
```
Comment (dinatale2) [inline .github/CONTRIBUTING.md]: Need to add that PRs should pass the automated testing.

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+[How To Contribute](#how-to-contribute)
+  * [Reporting a Bug](#reporting-a-bug)
+  * [Feature Suggestions](#feature-suggestions)
+  * [Pull Requests](#pull-requests)
+  * [License](#license)
+  * [Contributor's Declaration](#contributors-declaration)
+
+## Resources
+[mpiFileUtils Google Group](https://groups.google.com/forum/#!forum/mpifileutils)
```
Comment (dinatale2) [inline .github/CONTRIBUTING.md]: Do any other resources need to go here?

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
+# Contributing to mpiFileUtils
+
```
Comment (dinatale2) [inline .github/CONTRIBUTING.md]: Do we have a logo we can place at the top of this document?

File: .github/CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+*Thank you for taking the time to contribute!*
+
+## Table of Contents
+[Resources](#resources)
+
+[How To Contribute](#how-to-contribute)
+  * [Reporting a Bug](#reporting-a-bug)
```
Comment (dinatale2) [inline .github/CONTRIBUTING.md]: Need to fix the ToC.

File: .github/CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+### Pull Requests
+If you would like us to consider a change to mpiFileUtils, feel free to submit
+a pull request. Your pull request must meet the following criteria before before
+it can be merged.
+
+* All pull requests must be based on the current master branch and
+apply without conflicts.
+* Please try to limit pull requests to a single commit which resolves
+one issue.
+* Make sure your commit messages must have a `Signed-off-by` line. See
```
Comment (adammoody) [inline .github/CONTRIBUTING.md]: Looks like an extra "must" here?

**Comment by adammoody** [comment]
@dinatale2, thanks for putting this together.  Looks great.

**Comment by dinatale2** [comment]
@adammoody I went ahead and updated it. Do we have a logo or other resources we should provide?

**Comment by dinatale2** [comment]
Is this ok to merge? We can always make additions if the current version is acceptable.

**Comment by adammoody** [comment]
@dinatale2 Yes, looks good.  Let's merge.

**Pull request merged**